### ISSUE TITLE
Unexpected token f in JSON at position 1

### ISSUE THREAD

**Issue by NGuldager** [anvilco/spectaql#186]
Hi.

Seems like I'm getting a very similar issue as in https://github.com/anvilco/spectaql/issues/83, but it should've been resolved on 16 Sep 2021

I'm getting the following error:
```
➜  nestjs git:(staging) ✗ npx spectaql -d spectaql-config.yml
undefined:1
{first: 10}
 ^

SyntaxError: Unexpected token f in JSON at position 1
```

Schema: [schema.txt](https://github.com/anvilco/spectaql/files/7827743/schema.txt)

Config:

```
introspection:
  schemaFile: ./schema.gql

info:
  title: GraphQL API Reference
  description: Welcome to the party!
  termsOfService: https://www.example.com/terms
  contact:
    name: API Support
    url: http://www.example.com/support
    email: support@example.com
  license:
    name: Apache 2.0
    url: https://www.apache.org/licenses/LICENSE-2.0.html
```

Anything I'm doing wrong? Or anything I'm missing?

Full stack trace:
```
➜  nestjs git:(staging) ✗ npx spectaql -d spectaql-config.yml
undefined:1
{first: 10}
 ^

SyntaxError: Unexpected token f in JSON at position 1
    at JSON.parse (<anonymous>:null:null)
    at Object.resolveDefaultValue (/path/to/spectaql/node_modules/graphql-2-json-schema/dist/lib/reducer.js:64:21)
    at introspectionFieldReducer (/path/to/spectaql/node_modules/graphql-2-json-schema/dist/lib/reducer.js:38:46)
    at arrayReduce (/path/to/spectaql/node_modules/lodash/lodash.js:697:21)
    at Function.reduce (/path/to/spectaql/node_modules/lodash/lodash.js:9749:14)
    at introspectionFieldReducer (/path/to/spectaql/node_modules/graphql-2-json-schema/dist/lib/reducer.js:26:46)
    at arrayReduce (/path/to/spectaql/node_modules/lodash/lodash.js:697:21)
    at Function.reduce (/path/to/spectaql/node_modules/lodash/lodash.js:9749:14)
    at /path/to/spectaql/node_modules/graphql-2-json-schema/dist/lib/reducer.js:77:34
    at arrayReduce (/path/to/spectaql/node_modules/lodash/lodash.js:697:21)
    at Function.reduce (/path/to/spectaql/node_modules/lodash/lodash.js:9749:14)
    at Object.fromIntrospectionQuery (/path/to/spectaql/node_modules/graphql-2-json-schema/dist/lib/fromIntrospectionQuery.js:30:30)
    at jsonSchemaFromIntrospectionResponse (/path/to/spectaql/app/spectaql/graphql-loaders.js:90:20)
    at module.exports (/path/to/spectaql/app/spectaql/index.js:110:22)
    at loadData (/path/to/spectaql/app/index.js:172:95)
    at module.exports (/path/to/spectaql/app/index.js:176:80)
    at Object.<anonymous> (/path/to/spectaql/bin/spectaql.js:74:1)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
```

**Comment by newhouse** [comment]
Similar to #83 this looks to be an issue with `graphql-2-json-schema`. Could you open an [issue over there](https://github.com/charlypoly/graphql-to-json-schema/issues), please, and I'll have a look ASAP?

**Comment by jcuna** [comment]
I downgraded `graphql-2-json-schema` to 0.7.0 and 0.8.0 manually and issue still persists

**Comment by jcuna** [comment]
In my case, It actually works with a previous version of the `schema.gql` file. So there must be something that Apollo is actually adding to the output `gql` file that is not compatible with `graphql-2-json-schema`. All my tools currently mark the `.gql` file as valid. I'll keep digging and will update back.

**Comment by newhouse** [comment]
> In my case, It actually works with a previous version of the `schema.gql` file. So there must be something that Apollo is actually adding to the output `gql` file that is not compatible with `graphql-2-json-schema`. All my tools currently mark the `.gql` file as valid. I'll keep digging and will update back.

Can you provide the 2 files - 1 that works and 1 that does not?

I'm pretty sure that `graphql-to-json-schema` simply is not expecting / supporting what it's encountering because it's a bug there. This type of thing has come up before.

**Comment by jcuna** [comment]
> In my case, It actually works with a previous version of the `schema.gql` file. So there must be something that Apollo is actually adding to the output `gql` file that is not compatible with `graphql-2-json-schema`. All my tools currently mark the `.gql` file as valid. I'll keep digging and will update back.

I'm unable to show the schema as it's proprietary. I will continue to dig into it and provide relevant details as they become available.

**Comment by jcuna** [comment]
The problem I'm having arises from having default values for complex object types (non scalar). For example, the following syntax is a valid graphql syntax.
~~~gql
input SectionAttributesInput {
  render_on_page: Boolean = true
}

input NewSectionInput {
  archived: Boolean
  attributes: SectionAttributesInput = {render_on_page: true}
  ...
}
~~~
however, `graphql-2-json-schema.fromIntrospectionQuery`,  is unable to parse it with an error in the likes of `SyntaxError: Unexpected token r in JSON at position 1`

**Comment by jcuna** [comment]
I opened an issue over at graphql-to-json-schema https://github.com/charlypoly/graphql-to-json-schema/issues/67)(https://github.com/charlypoly/graphql-to-json-schema/issues/67

**Comment by newhouse** [comment]
Working on fix: https://github.com/charlypoly/graphql-to-json-schema/pull/69

**Comment by newhouse** [comment]
@jcuna fixed and published in `0.14.0`...thanks for your patience!

**Comment by newhouse** [comment]
cc @NGuldager

**Comment by jcuna** [comment]
> @jcuna fixed and published in `0.14.0`...thanks for your patience!

Thank you for tackling this. Appreciate your time.
